components/clientComponents/forms/Form/Form.tsx

```diff
@@ -189,9 +210,39 @@ const InnerForm: React.FC<InnerFormProps> = (props) => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [formStatusError, errorList, lastSubmitCount, canFocusOnError]);
 
-  const numberOfRequiredQuestions = form.elements.filter(
-    (element) => element.properties.validation?.required === true
-  ).length;
+  // Done in a callback to allow calling when the submit button is clicked. Otherwise the block
+  // below would be called very time the form values change. Similar reason why useMemo is not used.
+  const getNumberOfRequriedQuestions = () => {
```

Comment:
Let's look at how we can make this function testable via a unit test --passing in the params vs all the work being done inside.

i.e. I think the groupHistoryElements could be a stand-alone function and passed in.